### PR TITLE
sys-apps/likwid: bump to 4.3.1

### PR DESCRIPTION
**Pull request by brendanhoran**
Bug: https://bugs.gentoo.org/648380
Package-Manager: Portage-2.3.19, Repoman-2.3.6

### CODE REVIEW

File: sys-apps/likwid/files/likwid-4.3.1-Makefile.patch [hunk 1]
```diff
@@ -0,0 +1,21 @@
+--- likwid-likwid-4.3.1/Makefile.org	2018-02-26 13:59:28.421504363 +0800
```
Comment (mgorny) [inline sys-apps/likwid/files/likwid-4.3.1-Makefile.patch]: Please stop duplicating identical files. While at it, please prepend a commit removing existing duplicate files and updating 4.1.2 to use the files from 4.1.1.

File: sys-apps/likwid/likwid-4.3.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+LICENSE="GPL-3+"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="fortran"
+
+RDEPEND="dev-lang/perl"
+
+DEPEND="${RDEPEND}
+	fortran? ( sys-devel/gcc:*[fortran] )
```
Comment (mgorny) [inline sys-apps/likwid/likwid-4.3.1.ebuild]: Any reason not to use `virtual/fortran`?

File: sys-apps/likwid/likwid-4.3.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	"${FILESDIR}/${P}-config.mk.patch"
+)
+
+src_prepare() {
+	# Set PREFIX path to include sandbox path
+	sed -e 's:^PREFIX = .*:PREFIX = '${D}'/usr:' -i config.mk || \
+		die "Failed to set correct prefix path"
+
+	# Set the path to library directory.
+	sed -e 's:$(get_libdir):'$(get_libdir)':' -i config.mk || \
```
Comment (mgorny) [inline sys-apps/likwid/likwid-4.3.1.ebuild]: Please merge the two `sed`s on the same file. Also no need for `\` when the line ends with `||`.

File: sys-apps/likwid/likwid-4.3.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+		die "Cannot set library path!"
+
+	# Set correct LDFLAGS
+	sed -e '/LIBS/aSHARED_LFLAGS += -Wl,-soname,$@' \
+		-i make/include_GCC.mk || die "Failed to set correct LDFLAGS"
+
+	# Insert date and version info man pages
+	sed -e 's/<DATE>/21.08.2015/g' \
```
Comment (mgorny) [inline sys-apps/likwid/likwid-4.3.1.ebuild]: You sure about that date?

File: sys-apps/likwid/likwid-4.3.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+	# Set path to the access daemon, once installed into the system
+	sed -e "/exeprog/s|TOSTRING(ACCESSDAEMON)|\"/usr/sbin/likwid-accessD\"|" \
+		-i src/access_client.c || die "Failed to set access daemon path"
+
+	# Ensure we build with a non executable stack
+	sed -e "s:\$(SHARED_CFLAGS): \$(SHARED_CFLAGS)  -Wa,--noexecstack:" \
+	        -i make/config_defines.mk || die "Failed to set nonexecstack"
+
+	if use fortran; then
+
```
Comment (mgorny) [inline sys-apps/likwid/likwid-4.3.1.ebuild]: Please don't put empty line after `if`, indent is enough.

File: sys-apps/likwid/likwid-4.3.1.ebuild [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+	default
+
+}
+
+src_install () {
+	default
+	if use fortran; then
+		insinto /usr/include
+		doins likwid.mod
```
Comment (mgorny) [inline sys-apps/likwid/likwid-4.3.1.ebuild]: `doheader`?

File: sys-apps/likwid/files/likwid-4.3.1-Makefile.patch (hunk 1 above)
Comment (brendanhoran) [reply sys-apps/likwid/files/likwid-4.3.1-Makefile.patch]: Agree that  likwid-4.1.2-Makefile.patch likwid-4.1.1-Makefile.patch are identical, but likwid-4.3.1-Makefile.patch patches different lines in the Makefile. The patch contents is the same but where they occur inside the Makefile has changed. Hope that makes sense
How do you suggest I proceed here?

File: sys-apps/likwid/likwid-4.3.1.ebuild (hunk 4 above)
Comment (brendanhoran) [reply sys-apps/likwid/likwid-4.3.1.ebuild]: Will fix, pretty sure that is not right ;)

File: sys-apps/likwid/likwid-4.3.1.ebuild (hunk 3 above)
Comment (brendanhoran) [reply sys-apps/likwid/likwid-4.3.1.ebuild]: Sure, let me give that a try.

File: sys-apps/likwid/likwid-4.3.1.ebuild (hunk 6 above)
Comment (brendanhoran) [reply sys-apps/likwid/likwid-4.3.1.ebuild]: Sure, did not know about doheader. Will update this.

File: sys-apps/likwid/files/likwid-4.3.1-Makefile.patch (hunk 1 above)
Comment (mgorny) [reply sys-apps/likwid/files/likwid-4.3.1-Makefile.patch]: If it still applies, use the old patch.

**Comment by mgorny** [comment]
Small ping.

**Comment by mgorny** [comment]
If you are working on it, please also rebase it against current `master`, so that CI can test it.

**Comment by brendanhoran** [comment]
pushed new fixes and rebase

File: sys-apps/likwid/files/likwid-4.3.1-Makefile.patch (hunk 1 above)
Comment (brendanhoran) [reply sys-apps/likwid/files/likwid-4.3.1-Makefile.patch]: Old patch will not apply. Thus new patch needed.
I also after this version becomes stable want to drop all the older versions.
Simply not needed and will well with the patch file mess.

**Comment by brendanhoran** [comment]
I've also fixed up the other concerns around the sed statements and the date in the patch file. 
Hope its all good now, and sorry for the delay.

**Pull request closed without merging**